### PR TITLE
[WIP] Switch to Rust 2018

### PR DESCRIPTION
**Pull request by katyo**

### CODE REVIEW

**Comment by katyo** [comment]
@Drakulix I fixed code for compatibility with Rust 2018.
Also I eliminated double wrapping of interface (Rc<Box<I>> -> Rc<I>).

**Comment by katyo** [comment]
@Drakulix I think this PR is ready to merge.

File: src/context.rs [hunk 1]
```diff
@@ -168,21 +168,17 @@ impl Libinput {
         let context = unsafe {
             let udev = udev::udev_new();
             let libinput = ffi::libinput_udev_create_context(
-                &*boxed_interface as *const _,
-                &mut *boxed_userdata as *mut I as *mut libc::c_void,
+                Box::into_raw(boxed_interface),
+                Rc::as_ptr(&boxed_userdata) as *mut _,
```
Comment (Drakulix) [inline src/context.rs:172]: Neat! I didn't know about this function.

**Comment by Drakulix** [comment]
> @Drakulix I think this PR is ready to merge.

Indeed, looks nice. Thank you! :)

**Pull request merged**